riscv/src/compiler.rs

```diff
@@ -1014,10 +1020,13 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
             vec![format!("jump_and_link_dyn {rs};")]
         }
         "call" => {
-            if let [label] = args {
-                vec![format!("call {};", argument_to_escaped_symbol(label))]
-            } else {
-                panic!()
+            // Depending on what symbol is called, the call is replaced by a
+            // powdr asm call, or a call to a coprocessor if a special function
+            // has been recognized.
+            match args {
+                [Argument::Expression(Expression::Symbol(label))] if label == "poseidon_hash" => vec!["x0 <=X= poseidon(x10, x11);".to_string()],
```

Comment:
it seems that powder assembly uses `x10` and `x11` as the first two parameters of its calling convention. Let me know if this assumption is incorrect.